### PR TITLE
Add support for blobs in linera-storage::Storage

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

This is the first step for supporting user blobs (issue #1981)

## Proposal

Add blob support to `linera-storage::Storage`, and implement the functions.

## Test Plan

Will test this in the `PublishBlob` operation, but open to suggestions of testing specific for this PR, if you guys think it should be done

### CODE REVIEW

**Comment by andresilva91** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1984?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1984** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1984?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1965** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1965?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1964** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1964?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @andresilva91 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: linera-base/src/crypto.rs [hunk 1]
```diff
@@ -41,6 +41,17 @@ type HasherOutput = generic_array::GenericArray<u8, HasherOutputSize>;
 #[cfg_attr(with_testing, derive(Default))]
 pub struct CryptoHash(HasherOutput);
 
+/// A blob id.
+#[derive(Eq, PartialEq, Ord, PartialOrd, Clone, Copy, Hash, Debug, Serialize, Deserialize)]
+#[cfg_attr(with_testing, derive(Default))]
+pub struct BlobId(pub CryptoHash);
```
Comment (afck) [inline linera-base/src/crypto.rs:47]: This should probably be in `identifiers`, not `crypto`.

File: linera-base/src/crypto.rs [hunk 2]
```diff
@@ -41,6 +41,17 @@ type HasherOutput = generic_array::GenericArray<u8, HasherOutputSize>;
 #[cfg_attr(with_testing, derive(Default))]
 pub struct CryptoHash(HasherOutput);
 
+/// A blob id.
```
Comment (afck) [inline linera-base/src/crypto.rs:44]:
```suggestion
/// A blob ID.
```

File: linera-storage/src/db_storage.rs [hunk 3]
```diff
@@ -51,6 +51,17 @@ static CONTAINS_HASHED_CERTIFICATE_VALUE_COUNTER: Lazy<IntCounterVec> = Lazy::ne
     .expect("Counter creation should not fail")
 });
 
+/// The metric counting how often a blob is read from storage.
```
Comment (afck) [inline linera-storage/src/db_storage.rs:54]:
```suggestion
/// The metric counting how often a blob is tested for existence from storage.
```

File: linera-storage/src/db_storage.rs [hunk 4]
```diff
@@ -414,6 +466,12 @@ where
         self.write_batch(batch).await
     }
 
+    async fn write_blob(&self, blob: &Blob) -> Result<(), ViewError> {
```
Comment (christos-h) [inline linera-storage/src/db_storage.rs:469]: How about `write_blob` returning a `BlobId`?

**Review by ma2bd — APPROVED** [review]

**Pull request merged**